Tests/test_imageops.py

```diff
@@ -311,3 +313,30 @@ def autocontrast(cutoff):
 
         assert autocontrast(10) == autocontrast((10, 10))
         assert autocontrast(10) != autocontrast((1, 10))
+
+
+def test_autocontrast_mask():
+    # Test the mask argument of autocontrast
+    with Image.open("Tests/images/bw_gradient.png") as img:
+
+        rect_mask = Image.new("L", img.size, 0)
+        draw = ImageDraw.Draw(rect_mask)
+        x0, y0 = img.size[0]//4, img.size[1]//4
+        x1, y1 = 3*img.size[0]//4, 3*img.size[1]//4
+        draw.rectangle((x0, y0, x1, y1), fill=255)
+
+        assert ImageOps.autocontrast(img, mask=rect_mask) != ImageOps.autocontrast(img)
```

Comment:
This test asserts that an autocontrasted image with a mask is different an autocontrasted image without a mask. Okay.

But for all we know the masked image could be a blank black image. Could you add something to actually test for correctness of the masked output?

Reply:
Hmm. That will require some deeper thought. What you are suggesting is a functionality
test for the autocontrast method. 
I noticed that the baseline autocontrast did not have such a functional test in place.

1. So I guess the test would apply to both ?
2. I would likely have to add a custom test image for such a test. Any guidelines on this ?

How about we close this PR and follow up with the functional test in another PR ?

Reply:
We could test an arbitrary pixel is something expected. For example:

```diff
diff --git a/Tests/test_imageops.py b/Tests/test_imageops.py
index 3d5223d9..c5ad670a 100644
--- a/Tests/test_imageops.py
+++ b/Tests/test_imageops.py
@@ -342,3 +342,5 @@ def test_auto_contrast_mask_real_input():
         result_nomask = ImageOps.autocontrast(img)

         assert result_nomask != result
+        assert result.getpixel((20, 30)) == (239, 239, 242)
+        assert result_nomask.getpixel((20, 30)) == (132, 118, 96)
```

No need to close and create a new PR, we can update this one.

Reply:
Just added code to test for explicit pixel value in the test. Rather than look for a pixel value it compares the stats (median) of the autocontrast (with mask) and autocontrast (without mask). This logic draws from the theory of autocontrast and should scale well. PR 01aeaa4